wxQt: fails to link because function names are decorated with import library exports from Cairo.

### Build System Used

<!--

Please select the tool you use for building, i.e. check exactly
one line from those below by putting "x" in between the brackets.
-->

I build wxWidgets and/or my application using:

- [X] CMake
- [ ] configure
- [ ] mingw32-make with makefile.gcc
- [ ] MSBuild (Microsoft Visual Studio solution file)
- [ ] nmake with makefile.vc
- [ ] Xcode

<!--

For command line based tools, such as configure or CMake,
please include the full command line used to run them.

Please attach the full build log, but feel free to quote
the relevant parts of it here.

When using `configure`, please also attach `config.log` file.

-->

### Description

Testing the build will print this warning:
```
/usr/include/cairo/cairo.h:70:9: warning: "cairo_public" redefined
   70 | #define cairo_public _cairo_api extern
      |         ^~~~~~~~~~~~
src/common/cairo.cpp:19:9: note: this is the location of the previous definition
   19 | #define cairo_public
      |         ^~~~~~~~~~~~
```
and later the final link will fail with many messages like these ones:
```
[0/12] Linking CXX shared library lib/cygwx_qtu_core-3.2-0.dll
ld: libs/core/CMakeFiles/wxcore.dir/__/__/__/__/src/generic/graphicc.cpp.o: in function `wxCairoPathData::MoveToPoint(double, double)':
src/generic/graphicc.cpp:1293:(.text+0x1b7): undefined reference to `__imp_cairo_move_to'
ld: libs/core/CMakeFiles/wxcore.dir/__/__/__/__/src/generic/graphicc.cpp.o: in function `wxCairoPathData::AddLineToPoint(double, double)':
src/generic/graphicc.cpp:1298:(.text+0x1c7): undefined reference to `__imp_cairo_line_to'
ld: libs/core/CMakeFiles/wxcore.dir/__/__/__/__/src/generic/graphicc.cpp.o: in function `wxCairoPathData::~wxCairoPathData()':
src/generic/graphicc.cpp:1261:(.text+0x1e4): undefined reference to `__imp_cairo_destroy'
...etc
```
This error happens because the particular behaviour of DLLs.
This is how cairo handles the dllexport/dllimport modifiers:
https://cgit.freedesktop.org/cairo/tree/src/cairo.h#n53
```
#if (defined(_WIN32) || defined(__CYGWIN__)) && !defined(CAIRO_WIN32_STATIC_BUILD)
# define _cairo_export __declspec(dllexport)
# define _cairo_import __declspec(dllimport)
#elif defined(__GNUC__)
# define _cairo_export __attribute__((__visibility__("default")))
# define _cairo_import
#else
# define _cairo_export
# define _cairo_import
#endif

#ifdef CAIRO_COMPILATION
# define _cairo_api _cairo_export
#else
# define _cairo_api _cairo_import
#endif

#define cairo_public _cairo_api extern
```
So, declaring `cairo_public` doesn't avoid the use of dllexport/dllimport.
At the time of writing, I confirm that hacking the use of `CAIRO_WIN32_STATIC_BUILD` is the only way to build it.

However, it may be working in the past because a different behaviour of `cairo_public` into older versions of Cairo, so it must be inspected from which version this has changed and if such older versions are still able to be used by the current code.

### Platform and version information

- wxWidgets version <!-- [e.g. 3.2.1] --> you are building: 3.2.9
- wxWidgets port <!-- [e.g. wxMSW, wxGTK, wxOSX] --> you are building: wxQt
- OS <!-- [e.g. Windows 10, Ubuntu 22.10, macOS 15] --> and its version: CYGWIN/WIN32
- Compiler <!-- [e.g. MSVS 2022, gcc 12.1] --> being used: gcc-13.4.0
- Non-default compiler options, if any <!-- e.g. -std=c++26 -->: None.

--- a/include/wx/private/cairo.h
+++ b/include/wx/private/cairo.h
@@ -1,0 +1,29 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/private/cairo.h
+// Purpose:     Wrapper for including cairo.h from wx code
+// Author:      Vadim Zeitlin
+// Created:     2026-02-23
+// Copyright:   (c) 2026 Vadim Zeitlin <vadim@wxwidgets.org>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_PRIVATE_CAIRO_H_
+#define _WX_PRIVATE_CAIRO_H_
+
+// We want to prevent cairo.h from using dllimport attribute for the functions
+// it declares as we're loading them dynamically, but this is trickier than it
+// should be because older versions of cairo.h respected cairo_public being
+// defined before including it, but this was broken in 1.18, so we need to
+// define an alternative symbol and avoid defining cairo_public as this would
+// result in a warning about its redefinition.
+#include <cairo-version.h>
+
+#if CAIRO_VERSION_MAJOR > 1 || CAIRO_VERSION_MINOR >= 18
+#define CAIRO_WIN32_STATIC_BUILD
+#else
+#define cairo_public
+#endif
+
+#include <cairo.h>
+
+#endif // _WX_PRIVATE_CAIRO_H_

--- a/src/common/cairo.cpp
+++ b/src/common/cairo.cpp
@@ -14,11 +14,8 @@
 
 #if wxUSE_CAIRO && !defined(__WXGTK__)
 
-// keep cairo.h from defining dllimport as we're defining the symbols inside
-// the wx dll in order to load them dynamically.
-#define cairo_public
+#include "wx/private/cairo.h"
 
-#include <cairo.h>
 #include "wx/dynlib.h"
 
 #ifdef __WXMSW__

--- a/src/generic/graphicc.cpp
+++ b/src/generic/graphicc.cpp
@@ -16,13 +16,8 @@
 
 #if wxUSE_CAIRO
 
-#ifndef __WXGTK__
-// keep cairo.h from defining dllimport as we're defining the symbols inside
-// the wx dll in order to load them dynamically.
-#define cairo_public
-#endif
+#include "wx/private/cairo.h"
 
-#include <cairo.h>
 #include <float.h>
 
 bool wxCairoInit();

